sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -576,6 +576,12 @@ class Analyzer(
     /** Extracts a [[Generator]] expression and any names assigned by aliases to their output. */
     private object AliasedGenerator {
       def unapply(e: Expression): Option[(Generator, Seq[String])] = e match {
+        case Alias(g: Generator, name) if g.elementTypes.size > 1 => {
+          // In project, we probably give a default name which is not correct
+          // e.g. SELECT explode(map(key, value)) FROM src;
+          // Let's ignore the specified names for generator in project list
+          Some((g, Nil))
```

Comment:
The DataFrame API allows user to alias `explode` with 2 names like `df.select(explode($"myMap").as("key" :: "value" :: Nil))`, your change will break it.